src/frontend/src/catalog/index_catalog.rs

```diff
@@ -188,6 +189,26 @@ impl IndexCatalog {
         }
     }
 
+    pub fn get_column_def(&self, column_no: usize) -> Option<String> {
+        if let Some(col) = self.index_table.columns.get(column_no) {
+            if col.is_hidden {
+                return None;
+            }
+        } else {
+            return None;
+        }
+        let col = format!("{:?}", self.index_item[column_no]);
+        let column_names = self
+            .primary_table
+            .columns
+            .iter()
+            .map(|x| x.name())
+            .collect_vec();
+        // TODO(Kexiang): Currently we simply map $x to the column name. Extra info like ":Int32" is still kept.
+        // We should find a better way to generate the definition for `ExprImpl`.
```

Comment:
https://github.com/singularity-data/risingwave/blob/e0cfb82d96c281a4b5fa4114fc8064fae093c650/src/frontend/src/expr/mod.rs#L1033

Is this what you want?

Reply:
Good catch! It can help avoid a Regex replacement, but unfortunately cannot avoid the extra info like `:Int32`, which is hardcoded [here](https://github.com/risingwavelabs/risingwave/blob/95466f5be58dc4e93e851b1505db331234f83f6d/src/frontend/src/expr/literal.rs#L65). Since many of our existing implemention depends on this, I tend to keep it unchanged leave the mismatch for now. 

To erease the mismatch, we'd better introduce another wrapper, maybe be named as `ExprToDefinition` or `ExprDefinition`and similar to `ExprDisplay`, to rewrite the display functions for `ExprImpl`.